libraries/botbuilder-dialogs/src/dialogContainer.ts

```diff
@@ -96,8 +96,10 @@ export abstract class DialogContainer<O extends object = {}> extends Dialog<O> {
      * Future dialogs added to the component will also inherit this client.
      */
     public set telemetryClient(client: BotTelemetryClient) {
-        this._telemetryClient = client ? client : new NullTelemetryClient();
-        this.dialogs.telemetryClient = client;
+        this._telemetryClient = client ?? new NullTelemetryClient();
+        if (this.dialogs.telemetryClient !== this._telemetryClient) {
```

Comment:
Curious why this `if` check was added - it seems like any time this method runs, the result will be `this.dialogs.telemetryClient = this._telemetryClient`.

Reply:
If we don't check this, `set` method on `this.dialogs.telemetryClient` will always run, if a `DialogContainer`'s `dialogs` contains itself (since we are allowing cycle references), this method will run recursively and never end.